shap/explainers/_tree.py

```diff
@@ -501,6 +495,8 @@ def shap_values(self, X, y=None, tree_limit=None, approximate=False, check_addit
         if check_additivity and self.model.model_output == "raw":
             self.assert_additivity(out, self.model.predict(X))
 
+        if isinstance(out, list):
```

Comment:
Nit: suggest add a comment here explaining the interpretation of the various dimensions and why they are stacked.